packages/commerce-sdk-react/src/hooks/useMutation.ts

```diff
@@ -73,87 +72,49 @@ type MutationVariables = MutationVariablesWithBody | void
  */
 export const useCustomMutation = <TData = unknown, TError = unknown>(
     apiOptions: OptionalCustomEndpointClientConfig,
-    mutationOptions?: UseMutationOptions<TData, TError, MutationVariables>
-): UseMutationResult<TData, TError, MutationVariables> => {
+    mutationOptions?: UseMutationOptions<TData, TError, TMutationVariables>
+) => {
     const auth = useAuthContext()
-    const config = useConfig()
-    const clientHeaders = config.headers || {}
-    const currentClientConfig = {
+    const globalConfig = useConfig()
+    const globalHeaders = globalConfig.headers || {}
+    const globalClientConfig = {
         parameters: {
-            clientId: config.clientId,
-            siteId: config.siteId,
-            organizationId: config.organizationId,
-            shortCode: config.shortCode
+            clientId: globalConfig.clientId,
+            siteId: globalConfig.siteId,
+            organizationId: globalConfig.organizationId,
+            shortCode: globalConfig.shortCode
         },
-        proxy: config.proxy
+        proxy: globalConfig.proxy
     }
 
-    const callCustomEndpointWithAuth = (options: OptionalCustomEndpointClientConfig) => {
-        return async (): Promise<any> => {
-            const clientConfig = options.clientConfig || {}
+    const createMutationFnWithAuth = (): MutationFunction<TData, TMutationVariables> => {
+        return async (args): Promise<TData> => {
```

Comment:
The core change is in this line. We update the mutation function to take 
arguments.